src/main/java/org/symphonyoss/symphony/messageml/markdown/nodes/form/PersonSelectorNode.java

```diff
@@ -10,14 +10,42 @@
   private final static String MARKDOWN = "Person Selector";
 
   private String placeholder;
+  private String label;
+  private String tooltip;
 
-  public PersonSelectorNode(String placeholder) {
+  public PersonSelectorNode(String placeholder, String label, String tooltip) {
     super(MARKDOWN, placeholder);
     this.placeholder = placeholder;
+    this.label = label;
+    this.tooltip = tooltip;
   }
   
   @Override
   public String getText() {
-    return (placeholder != null) ? String.format(":[%s]", placeholder) : "";
+    StringBuilder markdownRepresentation = new StringBuilder();
+
+    if(placeholder != null || label != null || tooltip != null) {
+      markdownRepresentation.append(":");
+    }
+
+    if(placeholder != null) {
+      markdownRepresentation.append("[")
+          .append(placeholder)
+          .append("]");
+    }
+
+    if(label != null) {
```

Comment:
label/tooltip code is repeated 4 times. Maybe you could refactor by moving it into an interface and adding it into related classes. For e.g.:

```java

interface LabelTooltipNode {
  
default void appendLabelTooltip(StringBuilder markdownRepresentation){
  if(getLabel() != null){
   markdownRepresentation.append("[")
           .append(getLabel() )
           .append("]");
  }
  if(getTooltip() != null){
   markdownRepresentation.append("[")
           .append(getTooltip() )
           .append("]");
  }
}

public String getLabel();

public String getTooltip();
}

```

It is not only about duplicated code, but to be consistent and avoid mistakes while keeping the same order in label/tooltip markdown representation

Reply:
Done for Placeholder, Label and Tooltip